load-testing/tests/relays_stress_test.go

```diff
@@ -0,0 +1,457 @@
+
+	// Fund all the provisioned suppliers and gateways since their addresses are
+	// known and they are not created on the fly, while funding only the initially
+	// created applications.
+	fundedSuppliers, fundedGateways, fundedApplications := s.sendFundAvailableActorsTx(&plan)
+	// Funding messages are sent in a single transaction by the funding account,
+	// only one transaction is expected to be committed.
+	txResults := s.waitForTxsToBeCommitted()
+	s.ensureFundedActors(txResults, fundedSuppliers)
+	s.ensureFundedActors(txResults, fundedGateways)
+	s.ensureFundedActors(txResults, fundedApplications)
+
+	logger.Info().Msg("Actors funded")
+
+	// The initial actors are the first actors to stake.
+	suppliers := fundedSuppliers[:s.supplierInitialCount]
+	gateways := fundedGateways[:s.gatewayInitialCount]
+	applications := fundedApplications[:s.appInitialCount]
+
+	s.sendInitialActorsStakeMsgs(suppliers, gateways, applications)
+	txResults = s.waitForTxsToBeCommitted()
+	s.ensureStakedActors(txResults, MsgStakeSupplier, suppliers)
+	s.ensureStakedActors(txResults, MsgStakeGateway, gateways)
+	s.ensureStakedActors(txResults, MsgStakeApplication, applications)
+
+	logger.Info().Msg("Actors staked")
+
+	// Update the list of staked suppliers.
+	s.stakedSuppliers = append(s.stakedSuppliers, suppliers...)
+
+	// Delegate the initial applications to the initial gateways
+	s.sendDelegateInitialAppsTxs(applications, gateways)
+	txResults = s.waitForTxsToBeCommitted()
+	s.ensureDelegatedApps(txResults, applications, gateways)
+
+	logger.Info().Msg("Apps delegated")
+
+	// Applications and gateways are now ready and will be active in the next session.
+	s.preparedApplications = append(s.preparedApplications, applications...)
+	s.preparedGateways = append(s.preparedGateways, gateways...)
+
+	// batchInfoObs maps session information to batch information used to schedule
+	// the relay requests to be sent on the current block.
+	batchInfoObs, batchInfoPublishCh := channel.NewReplayObservable[*batchInfoNotif](s.ctx, 5)
+	s.batchInfoObs = batchInfoObs
+
+	// sessionInfoObs asynchronously maps committed blocks to a notification which
+	// includes the session number and the start and end block heights of the session.
+	// It runs at the same frequency as committed blocks (i.e. 1:1).
+	s.sessionInfoObs = channel.Map(
+		s.ctx,
+		s.blockClient.CommittedBlocksSequence(s.ctx),
+		s.mapSessionInfoFn(batchInfoPublishCh),
+	)
+
+	// stakingObs asynchronously maps session information to a set of newly staked
+	// actor accounts, only notifying when new actors were staked and skipping otherwise.
+	// It stakes new suppliers & gateways but only funds new applications as they can't be
+	// delegated until after the respective gateway stake txs have been committed.
+	// It receives at the same frequency as committed blocks (i.e. 1:1) but only sends
+	// conditionally as described here.
+	stakingObs := channel.Map(s.ctx, s.sessionInfoObs, s.mapStakingInfoFn(plan))
```

Comment:
NIT: Why did you factor `mapStakingInfoFn` out into a separate function but define the other two in place?